src/Tokenizer/IndexNotFoundException.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tokenizer;
+
+/**
+ * @author Gregor Harlan
+ *
+ * @no-named-arguments Parameter names are not covered by the backward compatibility promise.
+ */
+final class IndexNotFoundException extends \RuntimeException
+{
+    private function __construct(string $message)
+    {
+        parent::__construct($message);
+    }
+
+    public static function fromMethod(string $methodName): self
```

Comment:
[ref](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/9008#discussion_r2315685509)

at least 
```suggestion
    /** @internal */
    public static function fromMethod(string $methodName): self
```